concrete/controllers/dialog/type/update_from_type.php

```diff
@@ -0,0 +1,273 @@
+              $addedChildPageblock = \Block::getByID($pageTypeBlock->getBlockID(), $childPage, $blockToAdd['pageTypeArHandle']);
+              $addedChildPageblock->setAbsoluteBlockDisplayOrder($blockToAdd['actualDisplayOrder']);
+            }
+
+            foreach ($blocksToUpdate as $blockToUpdate) {
+              $childPageBlock = \Block::getByID($blockToUpdate['bID'], $childPage, $blockToUpdate['arHandle']);
+              $childPageBlockPerms = new \Permissions($childPageBlock);
+
+              if (!is_object($childPageBlock) || $childPageBlock->isError() || !$childPageBlockPerms->canAdminBlock()) {
+                continue;
+              }
+
+              $actions = $blockToUpdate['actions'];
+
+              foreach ($actions as $action) {
+                if($action['name'] == 'update_forked'){
+                  $pageTypeBlock = \Block::getByID($action['pageTypeBlockID'], $pageTypeDefaultPage, $action['pageTypeArHandle']);
+                  $bt = $pageTypeBlock->getBlockTypeObject();
+
+                  $childPageBlock->deleteBlock();
+
+                  if ($bt->isCopiedWhenPropagated()) {
+                    $childPageBlock = $pageTypeBlock->duplicate($childPage, true);
+                  }else{
+                    $pageTypeBlock->alias($childPage);
+                  }
+                }
+                else if ($action['name'] == 'change_arHandle') {
+
+                  $nh = $action['actualArHandle'];
+                  $pageCollectionID = $childPage->getCollectionID();
+                  $pageVersionID = $childPage->getVersionID();
+
+                  $db->executeQuery('UPDATE CollectionVersionBlockStyles SET arHandle = ?  WHERE cID = ? and cvID = ? and bID = ?',
+                               [$nh, $pageCollectionID, $pageVersionID, $childPageBlock->getBlockID()]);
+                  $db->executeQuery('UPDATE CollectionVersionBlocks SET arHandle = ?  WHERE cID = ? and cvID = ? and bID = ?',
+                               [$nh, $pageCollectionID, $pageVersionID, $childPageBlock->getBlockID()]);
+                }
+                else if ($action['name'] == 'change_display_order') {
+                  $childPageBlock->setAbsoluteBlockDisplayOrder($action['actualDisplayOrder']);
+                  array_push($handlesToOrder, $action['actualArHandle']);
+                }
+                else if ($action['name'] == 'delete') {
+                  $childPageBlock->deleteBlock();
+                }
+              }
+
+            }
+
+            foreach ($handlesToOrder as $handleToOrder) {
+              $childPage->rescanDisplayOrder($handleToOrder);
+            }
+
+            $queue->deleteMessage($message);
+          }
+
+          $obj->totalItems = $queue->count();
+
+          if ($queue->count() == 0) {
+              $queue->deleteQueue($name);
+          }
+
+          $obj->message = t('All child child pages updated successfully.');
+          echo json_encode($obj);
```

Comment:
@KorvinSzanto tnx and sorry for big delay - I was AFK on my vacation.

- Can you give an example ResponseFactory and json, or point to some docs? Please.

- Why it's OK to use `echo json_encode($obj)` in this script https://github.com/concrete5/concrete5/blob/develop/concrete/controllers/dialog/block/aliasing.php, not in mine :) ?